include/swift/AST/SourceFile.h

```diff
@@ -142,6 +142,7 @@ class SourceFile final : public FileUnit {
 
   /// If non-null, used to track name lookups that happen within this file.
   Optional<ReferencedNameTracker> ReferencedNames;
+  Optional<ReferencedNameTracker> RequestReferencedNames;
```

Comment:
Now I'm thinking that the ReferencedNameTracker was passed separately into the dependency creation code in order to allow you to pass either one of these. Is that right? If so, I'm uneasy because passing a separate pointer destroys the invariant that the tracker passed in belongs to the SourceFile. I would rather pass in a bool to the dependency creation code than a separate pointer.